### PR TITLE
new CalculateStagingAreaMatchablesScheduledTask to recalculate matchables

### PR DESCRIPTION
**Pull request by ChemMitch**
for the staging area

### CODE REVIEW

File: gsrs-spring-boot-autoconfigure/src/main/java/gsrs/tasks/CalculateStagingAreaMatchablesScheduledTask.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        threadCount=count;
+    }
+
+    private final Map<String, StagingAreaService> servicesByContext = new HashMap<>();
+
+    private StagingAreaService getStagingAreaService(String contextName) {
+        String contextNameToUse =contextName;
+        if(contextNameToUse.contains(".")) {
+            String[] parts =contextNameToUse.split("\\.");
+            contextNameToUse = parts[parts.length-1].toLowerCase() + "s";
```
[the remark this answers is not in the document]
Comment (tylerperyea) [reply gsrs-spring-boot-autoconfigure/src/main/java/gsrs/tasks/CalculateStagingAreaMatchablesScheduledTask.java:64]: While the API endpoint (context) is usually the pluralized lower-case version of the root entity class name, I don't think this is a safe assumption in every case.

[the remark this answers is not in the document]
Comment (ChemMitch) [reply gsrs-spring-boot-autoconfigure/src/main/java/gsrs/tasks/CalculateStagingAreaMatchablesScheduledTask.java:64]: Added EntityContextLookup as discussed

File: gsrs-spring-boot-autoconfigure/src/main/java/gsrs/tasks/CalculateStagingAreaMatchablesScheduledTask.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        }
+        log.trace("supplied contextName: {}, used {}", contextName, contextNameToUse);
+        return servicesByContext.computeIfAbsent(contextNameToUse, n-> {
+            Class stagingAreaServiceClass=importAdapterFactoryFactory.getDefaultStagingAreaService(n);
+            Constructor<StagingAreaService> constructor= null;
```
[the remark this answers is not in the document]
Comment (ChemMitch) [reply gsrs-spring-boot-autoconfigure/src/main/java/gsrs/tasks/CalculateStagingAreaMatchablesScheduledTask.java:69]: Commented it out, replacing it with 
`servicesByContext.get(contextNameToUse)`
and got NPEs with each record

File: gsrs-spring-boot-autoconfigure/src/main/java/gsrs/tasks/CalculateStagingAreaMatchablesScheduledTask.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+    private StagingAreaService getStagingAreaService(String contextName) {
+        String contextNameToUse =contextName;
+        if(contextNameToUse.contains(".")) {
+            String[] parts =contextNameToUse.split("\\.");
+            contextNameToUse = parts[parts.length-1].toLowerCase() + "s";
+        }
+        log.trace("supplied contextName: {}, used {}", contextName, contextNameToUse);
+        return servicesByContext.computeIfAbsent(contextNameToUse, n-> {
+            Class stagingAreaServiceClass=importAdapterFactoryFactory.getDefaultStagingAreaService(n);
```
[the remark this answers is not in the document]
Comment (ChemMitch) [reply gsrs-spring-boot-autoconfigure/src/main/java/gsrs/tasks/CalculateStagingAreaMatchablesScheduledTask.java:68]: ibid

**Review by tylerperyea — APPROVED** [review]

**Pull request merged**